subsys/debug/asan.c

```diff
@@ -16,3 +21,14 @@ const char *__asan_default_options()
 #endif
 		"";
 }
+
+#if defined(CONFIG_SOC_POSIX)
+
+static void lsan_do_leak_check(void)
+{
+	__lsan_do_leak_check();
```

Comment:
Will this bite us back due to this interface not being stable?

Reply:
Wasn't this necessary only for 64 bit targets?
(Not having looked into ASAN leak detection) Note that leak checks are not really useful anyhow for Zephyr on native_posix, as Zephyr's heap allocation is not using the Linux heap.
Leak detection can help you only with HW models or native_posix drivers leaks, but at this point (NATIVE_TASK ON_EXIT) there can still be some memory allocated by the board / HW models.
Meaning, would it maybe be easier to just disable ASAN leaks detection always?

Reply:
> Will this bite us back due to this interface not being stable?

The interface has been there for a while and is both implemented by GCC and LLVM, so I assume it will not disappear quickly.

> Wasn't this necessary only for 64 bit targets?

Actually it is only need for 64-bit when building with GCC, but I rather keep the behavior the same for GCC and LLVM. Note that leak detection is not supported in 32-bit mode by ASAN.

>  Note that leak checks are not really useful anyhow for Zephyr on native_posix, as Zephyr's heap allocation is not using the Linux heap.

This is not 100% correct, a zephyr application can use malloc/free from libc and on a posix arch board this will result in the use of linux's malloc/free.

> Leak detection can help you only with HW models or native_posix drivers leaks, but at this point (NATIVE_TASK ON_EXIT) there can still be some memory allocated by the board / HW models.

This is true but we could add a suppression file.

> Meaning, would it maybe be easier to just disable ASAN leaks detection always?

I rather keep this functionality as zephyr apps could use malloc/free through libc.